### PR TITLE
suggestions for compare_branches

### PR DESCRIPTION
**Pull request by fabian-s**
I had to fiddle around a little bit with this to get it to work for #397, maybe you'll find these useful/sensible as well:  

- 105: make sure `if(is.na(...`  always evaluates to `logical(1)`
- 153: actually use parameter "sample_size"
- 349f: more fault-tolerant loop over linters
- 370: flush warnings to terminal at the end

### CODE REVIEW

**Comment by fabian-s** [comment]
Also: If you ask pull request makers to run this to evaluate a new linter they propose, I think they also need to define a "dummy" linter of the same name in `master`, like:

```
multiple_dot_args_linter <- function(source_file) {
   Linter(function(source_file) return(list())))
}
```

**Comment by MichaelChirico** [comment]
> I think they also need to define a "dummy" linter of the same name in master

I'd rather refactor the code not to run anything, even with a decoy linter it'll take a long time to run if nothing's accomplished

File: .dev/compare_branches.R [hunk 1]
```diff
@@ -346,10 +346,18 @@ if (length(packages) > 50L) {
   )
 }
 
+df_otherwise <- tibble::tibble(
```
Comment (MichaelChirico) [inline .dev/compare_branches.R:349]: could you use tidyverse style here?

https://style.tidyverse.org/syntax.html#long-lines

Comment (fabian-s) [reply .dev/compare_branches.R:359]: sorry. see https://github.com/r-lib/lintr/pull/982/commits/1798de0810868a5c7c09c5e84d2962661dc022c9

**Comment by MichaelChirico** [comment]
Thanks for the improvements!!

File: .dev/compare_branches.R (hunk 1 above)
Comment (MichaelChirico) [reply .dev/compare_branches.R:359]: np. understand styler::style is the easiest approach for such a request. and we should probably comply with that globally anyway.

**Pull request merged**